controller/config.go

```diff
@@ -861,6 +862,12 @@ func validateCAASImageRepo(imageRepo string) (string, error) {
 		return "", errors.Trace(err)
 	}
 	defer func() { _ = r.Close() }()
+
+	if err = r.Ping(); registryutils.IsPublicAPINotAvailableError(err) {
+		logger.Warningf("%v, upgrade controller/model will not work, please consider to provide registry credential", err)
```

Comment:
I think this would be a cleaner error message.
`logger.Warningf("docker registry for %q requires authentication: %v", imageRepo, err)`